mita/openstack.py

```diff
@@ -147,15 +147,27 @@ def _get_image(self) -> NodeImage:
             logger.error(be)
             raise OpenStackDriverError("Encountered an unknown exception.")
 
-    def _get_flavor(self) -> NodeSize:
-        """Return the flavor reference."""
-        try:
-            return [f for f in self.driver_v2.list_sizes() if f.name == self.vm_size][0]
-        except IndexError:
-            raise ResourceNotFound("Flavor {} not found".format(self.vm_size))
-        except BaseException as be:  # noqa
-            logger.error(be)
-            raise OpenStackDriverError("Encountered an unknown exception.")
+    def _get_flavor(self, name: Optional[str] = None) -> NodeSize:
+        """
+        Return the NodeSize instance retrieved based on the provided name or vm_size.
+
+        Args:
+            name:   The name of the flavor to be retrieved
```

Comment:
It would be great to provide an example here, like `m1.large` or whatever we expect